spin-observatory-plugin-deck/src/components/PluginContainer.tsx

```diff
@@ -17,6 +17,7 @@ export function PluginContainer({ app }: IPluginContainerProps) {
   const { data: pipelines } = useDataSource<IPipeline[]>(dataSource);
   const [selectedPipeline, setSelectedPipeline] = useState<IPipeline>();
   const [selectedParams, setSelectedParams] = useState<string[]>([]);
+  const [selectedDateFilter, setSelectedDateFilter] = useState<IDateFilter>();
```

Comment:
You can pass in a default value for the `selectedDateFilter` state like this `useState<IDateFilter>({ start: 0, end: MAX_DATE_FILTER });` and I think this will let you avoid the `if (!dateFilter)` check later on

Reply:
Nice! Removed the two checks.